main.tf

```diff
@@ -40,28 +40,45 @@ module "subnet_config" {
 
 resource "azurerm_kubernetes_cluster" "aks" {
   depends_on          = [azurerm_role_assignment.route_table_network_contributor]
+  #lifecycle {
+  #  ignore_changes = [(local.node_pools[var.default_node_pool].enable_auto_scaling ? "node_count" : "")] 
+  #}
+
   name                = local.cluster_name
   location            = var.location
   resource_group_name = var.resource_group_name
   dns_prefix          = "${var.names.product_name}-${var.names.environment}-${var.names.location}"
   tags                = var.tags
 
   kubernetes_version = var.kubernetes_version
-  
+
   network_profile {
     network_plugin = var.network_plugin
   }
 
   default_node_pool {
-    name                = var.default_node_pool_name
-    vm_size             = var.default_node_pool_vm_size
-    enable_auto_scaling = var.default_node_pool_enable_auto_scaling
-    node_count          = (var.default_node_pool_enable_auto_scaling ? null : var.default_node_pool_node_count)
-    min_count           = (var.default_node_pool_enable_auto_scaling ? var.default_node_pool_node_min_count : null)
-    max_count           = (var.default_node_pool_enable_auto_scaling ? var.default_node_pool_node_max_count : null)
-    availability_zones  = var.default_node_pool_availability_zones
-    vnet_subnet_id      = (var.aks_managed_vnet ? null : var.node_pool_subnets[var.default_node_pool_subnet].id)
-    tags                = var.tags
+    name                         = local.node_pools[var.default_node_pool].name
+    vm_size                      = local.node_pools[var.default_node_pool].vm_size
+    os_disk_size_gb              = local.node_pools[var.default_node_pool].os_disk_size_gb
+    os_disk_type                 = local.node_pools[var.default_node_pool].os_disk_type
+    availability_zones           = local.node_pools[var.default_node_pool].availability_zones
+    enable_auto_scaling          = local.node_pools[var.default_node_pool].enable_auto_scaling
```

Comment:
Would it be valuable to add defaults to any of theses?

Reply:
they automatically inherit the values in the variable node_pool_defaults